internal/deployment_client.go

```diff
@@ -0,0 +1,244 @@
+		// UpsertEntries - Metadata entries inserted or modified. When values are not
+		// of type *commonpb.Payload, the client data converter will be used to generate
+		// payloads.
+		UpsertEntries map[string]interface{}
+
+		// RemoveEntries - List of keys to remove from the metadata.
+		RemoveEntries []string
+	}
+
+	// DeploymentSetCurrentOptions provides options for [DeploymentClient.SetCurrent].
+	// NOTE: Experimental
+	DeploymentSetCurrentOptions struct {
+		// Deployment - An identifier for this deployment.
+		Deployment Deployment
+
+		// MetadataUpdate - Optional: Changes to the user-defined metadata entries
+		// for this deployment.
+		MetadataUpdate DeploymentMetadataUpdate
+	}
+
+	// DeploymentSetCurrentResponse is the response type for [DeploymentClient.SetCurrent].
+	// NOTE: Experimental
+	DeploymentSetCurrentResponse struct {
+		// Current - Information about the current deployment after this operation.
+		Current DeploymentInfo
+
+		// Previous - Information about the last current deployment, i.e., before this operation.
+		Previous DeploymentInfo
+	}
+
+	// DeploymentDescribeOptions provides options for [DeploymentClient.Describe].
+	// NOTE: Experimental
+	DeploymentDescribeOptions struct {
+		// Deployment - Identifier that combines the deployment series name with their Build ID.
+		Deployment Deployment
+	}
+
+	// DeploymentGetReachabilityOptions provides options for [DeploymentClient.GetReachability].
+	// NOTE: Experimental
+	DeploymentGetReachabilityOptions struct {
+		// Deployment - Identifier that combines the deployment series name with their Build ID.
+		Deployment Deployment
+	}
+
+	// DeploymentGetCurrentOptions provides options for [DeploymentClient.GetCurrent].
+	// NOTE: Experimental
+	DeploymentGetCurrentOptions struct {
+		// SeriesName - Name of the deployment series.
+		SeriesName string
+	}
+
+	// DeploymentClient is the client that manages deployments.
+	// NOTE: Experimental
+	DeploymentClient interface {
+		// Describes an existing deployment.
+		// NOTE: Experimental
+		Describe(ctx context.Context, options DeploymentDescribeOptions) (DeploymentInfo, error)
+
+		// List returns an iterator to enumerate deployments in the client's namespace.
+		// It can also optionally filter deployments by series name.
+		// NOTE: Experimental
+		List(ctx context.Context, options DeploymentListOptions) (DeploymentListIterator, error)
+
+		// GetReachability returns reachability information for a deployment. This operation is
+		// expensive, and results may be cached. Use the returned [DeploymentReachabilityInfo.LastUpdateTime]
+		// to estimate cache staleness.
+		// NOTE: Experimental
+		GetReachability(ctx context.Context, options DeploymentGetReachabilityOptions) (DeploymentReachabilityInfo, error)
+
+		// GetCurrent returns the current deployment for a given deployment series.
+		// NOTE: Experimental
+		GetCurrent(ctx context.Context, options DeploymentGetCurrentOptions) (DeploymentInfo, error)
```

Comment:
How confident are you that we will never add anything else to `GetCurrentDeploymentResponse`?

Reply:
Same argument as above, I wanted it to be symmetrical with `Describe`,  I think we can extend the struct `DeploymentInfo` without needing a wrapper...

Reply:
We often add things to the gRPC response message and not the single info field within. We have done this before on other calls and regret reusing types as the results on these methods because we are forced to make more.

Reply:
OK, wrapping the responses